File: Java/src/com/dbc/algorithm/Daily/MinCostClimbingStairs.java
package com.dbc.algorithm.Daily;

public class MinCostClimbingStairs {
    public static int minCostClimbingStairs(int[] cost) {
        int length = cost.length;
        for (int i = 2; i < length; i++) {
            cost[i] += cost[i - 1] >= cost[i - 2] ? cost[i - 2] : cost[i - 1];
        }
        return cost[length - 1] < cost[length - 2] ? cost[length - 1] : cost[length - 2];
    }
}

File: Python/min_cost_climbing_stairs.py
class Solution(object):
    @staticmethod
    def min_cost_climbing_stairs(cost):
        """
        动态规划解法
        :type cost: List[int]
        :rtype: int
        """
        for i in range(2, len(cost)):
            cost[i] = min(cost[i] + cost[i - 1], cost[i] + cost[i - 2])
                
        return min(cost[-1], cost[-2])

if __name__ == "__main__":
    print(Solution.min_cost_climbing_stairs(cost=[1, 100, 1, 1, 1, 100, 1, 1, 100, 1]))